### ISSUE TITLE
New Menüitem-Type Module

### ISSUE THREAD

**Issue by designbengel** [Digital-Peak-Incubator/tpl_tauristar#88]
For having an extended menu later it would be good to have a menuitem-type module

**Comment by laoneo** [comment]
Can you extend the description a bit, can't understand what you mean. If possible add some screenshots.

**Comment by laoneo** [comment]
Any comment on this one?

**Comment by designbengel** [comment]
Have no time to create a mockup but maybe to explain better. 
There are some menutypes like component, heading, seperator, Url
i would add one for module to be able to set that a module should appear inside the menu - Hope that was more clear?

**Comment by laoneo** [comment]
This request is not within the scope of this project as it is a new feature of the cms. I suggest to open an issue on the cms tracker.

**Comment by designbengel** [comment]
Ok, i thought we are doing PRs to the cms in favour for the template?

**Comment by laoneo** [comment]
If there are some issues with bs3 compatibility, then we will do them, but mainly when we found some hardcoded HTML in PHP files which should be moved to a JLayout, then we are porting them back to the CMS. The only new feature of this project should be a new template with all the bs3 and scss stuff, including some helper code as mentioned in #95. New features which belong to the CMS and not this project should be opened in the issue tracker.
Imagine, at the end this project will be one single pr against the CMS repository, when we start to bring in some more stuff, it will be almost impossible to review it then by the maintainers. Better to open new feature issues which do come up in this project directly on the CMS. You will get better feedback there.

**Comment by designbengel** [comment]
But hard to get then a decent Mega-Menu for the new template then?

**Comment by laoneo** [comment]
Will the mega menu be loaded on the start page only? If yes, a solution can be to load a module within the startpage article. Perhaps I miss here something, but from the technical perspective, the issue belongs to the CMS repo. If it will get accepted there, we can always merge it back to this repo, if not, then it is also unlikely that we should include it in this project as well.

**Comment by designbengel** [comment]
No not only on the start page. I mean a menu with multiple columns and possibility to place modules there. How can we achieve this in the template?

**Comment by laoneo** [comment]
I guess now I need some screenshots or links what you want.

**Comment by designbengel** [comment]
Something like these examples here: https://www.google.de/search?q=mega-menu&espv=2&biw=1516&bih=907&source=lnms&tbm=isch&sa=X&ved=0ahUKEwjgxbD-oJHOAhWG6CwKHRfFAfcQ_AUIBigB

**Comment by laoneo** [comment]
Guess that's possible. The only question I have is how to place the modules? What for a position to give them?

**Comment by designbengel** [comment]
you can give the module any positionname and call it then in the menu with the customname...

**Comment by laoneo** [comment]
But then it is not dynamic, then you can place the modules only in the hard coded columns.

**Comment by designbengel** [comment]
Why not dynamic? You can define a moduleposition on the fly it has been a long time like that, and then you can choose which moduleposition you want to load in the menue? But i think you need to need a new menu-item-type then, or how to solve it different?

**Comment by laoneo** [comment]
A google search for Joomla mega menu offered me some already existing menus, should we not use such one instead making our own?

**Comment by designbengel** [comment]
yes we can use an existing one, but we need the possibility for the settings

**Comment by laoneo** [comment]
I'v played a bit around with it and I think this definitely belongs to the the CMS repository as it needs a rewrite of the mod_menu module. Can you please open an issue there and link it to this one here.
